README.md

```diff
@@ -1,2 +1,981 @@
+
+#### Defining libraries using _config.cmake_
+
+The set of `.c` files that will compose each library shall be enough to cover all APIs exposed by the header _pal.h_, in our case, it needs one adapter for the temperature sensor, fan, led, exception, and thread. The definition of these configurations are in the _config.cmake_. A _.cmake_ file is a file that can be included in any _CMakeLists.txt_ to export variables or functions, in this sample, the _config.cmake_ expose the list of configurations and the functions to create samples and tests. 
+
+Observe that we defined all possible configurations in the _config.cmake_, independent of the OS. As a convention, we define configurations pal_win_x for windows, and pal_linux_x for windows 
+
+```cmake
+##
+# Description of PAL configurations for Windows.
+##
+list(APPEND pal_win_1
+    pal_version.c
+    pal_temperature_a.c
+    pal_led.c
+    pal_fan_a.c
+    pal_exception.c
+    pal_thread_windows.c
+)
+
+list(APPEND pal_win_2
+    pal_version.c
+    pal_temperature_b.c
+    pal_led.c
+    pal_fan_b.c
+    pal_exception.c
+    pal_thread_windows.c
+)
+
+list(APPEND pal_win_3
+    pal_version.c
+    pal_temperature_b.c
+    pal_led.c
+    pal_fan_c.c
+    pal_exception.c
+    pal_thread_windows.c
+)
+
+##
+# Description of PAL configurations for Linux.
+##
+list(APPEND pal_linux_1
+    pal_version.c
+    pal_temperature_a.c
+    pal_led.c
+    pal_fan_a.c
+    pal_exception.c
+    pal_thread_pthread.c
+)
+
+list(APPEND pal_linux_2
+    pal_version.c
+    pal_temperature_b.c
+    pal_led.c
+    pal_fan_b.c
+    pal_exception.c
+    pal_thread_pthread.c
+)
+
+list(APPEND pal_linux_3
+    pal_version.c
+    pal_temperature_b.c
+    pal_led.c
+    pal_fan_c.c
+    pal_exception.c
+    pal_thread_pthread.c
+)
+```
+
+To define each configurations are valid for each OS, we created a list called `pal_libraries` that contains all libraries that the cmake shall create and test.
+
+Cmake has a few build variables, a full documentation can be find at https://cmake.org/documentation/. For the propose of this sample, we will test if the build variable `WIN32` was defined or not (as you can presume, this variable only exist if you are running the cmake in a Windows machine).
+
+```cmake
+if(WIN32)
+    list(APPEND pal_libraries 
+        pal_win_1 
+        pal_win_2
+        pal_win_3
+    )
+else()
+    list(APPEND pal_libraries 
+        pal_linux_1 
+        pal_linux_2
+        pal_linux_3
+    )
+endif()
+```
+
+So if `WIND32` is defined, `pal_libraries` will contain _pal_win_1_, _pal_win_2_, and _pal_win_3_, if not, this variable will contain _pal_linux_1_, _pal_linux_2_, and _pal_linux_3_. Other configurations may be necessary, for example, if you are using Clang as your compiler, you will need to make a more complex decision, because Clang uses pthread library on both Linux and Windows. If it is the case, the reliable way is to use the cmake variable `CMAKE_C_COMPILER_ID`.
```

Comment:
```suggestion
So if `WIN32` is defined, `pal_libraries` will contain _pal_win_1_, _pal_win_2_, and _pal_win_3_, if not, this variable will contain _pal_linux_1_, _pal_linux_2_, and _pal_linux_3_. Other configurations may be necessary, for example, if you are using Clang as your compiler, you will need to make a more complex decision, because Clang uses pthread library on both Linux and Windows. If it is the case, the reliable way is to use the cmake variable `CMAKE_C_COMPILER_ID`.
```